examples/js/loaders/GLTFLoader.js

```diff
@@ -2965,25 +2962,58 @@ THREE.GLTFLoader = ( function () {
 
 					return node;
 
-				} );
+				} ) );
+
+			}
 
-			} else if ( nodeDef.camera !== undefined ) {
+			if ( nodeDef.camera !== undefined ) {
 
-				return parser.getDependency( 'camera', nodeDef.camera );
+				pending.push( parser.getDependency( 'camera', nodeDef.camera ) );
 
-			} else if ( nodeDef.extensions
+			}
+
+			if ( nodeDef.extensions
 				&& nodeDef.extensions[ EXTENSIONS.KHR_LIGHTS_PUNCTUAL ]
 				&& nodeDef.extensions[ EXTENSIONS.KHR_LIGHTS_PUNCTUAL ].light !== undefined ) {
 
-				return parser.getDependency( 'light', nodeDef.extensions[ EXTENSIONS.KHR_LIGHTS_PUNCTUAL ].light );
+				pending.push( parser.getDependency( 'light', nodeDef.extensions[ EXTENSIONS.KHR_LIGHTS_PUNCTUAL ].light ) );
+
+			}
+
+			return Promise.all( pending );
+
+		}() ).then( function ( objects ) {
+
+			var node;
+
+			// .isBone isn't in glTF spec. See .markDefs
+			if ( nodeDef.isBone === true ) {
+
+				node = new THREE.Bone();
+
+			} else if ( objects.length > 1 ) {
+
+				node = new THREE.Group();
+
+			} else if ( objects.length === 1 ) {
+
+				node = objects[0];
 
 			} else {
 
-				return Promise.resolve( new THREE.Object3D() );
+				node = new THREE.Object3D();
 
 			}
 
-		}() ).then( function ( node ) {
+			if (node !== objects[0]) {
```

Comment:
Minor: could you run the linter on this file? There are a few spots that should have more whitespace, `( node !== objects[ 0 ] )`.